internal/controller/helmchart_controller.go

```diff
@@ -536,7 +539,7 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		// this is needed because otherwise the credentials are stored in ~/.docker/config.json.
 		// TODO@souleb: remove this once the registry move to Oras v2
 		// or rework to enable reusing credentials to avoid the unneccessary handshake operations
-		registryClient, credentialsFile, err := r.RegistryClientGenerator(clientOpts.RegLoginOpt != nil)
+		registryClient, credentialsFile, err := r.RegistryClientGenerator(clientOpts.TlsConfig, clientOpts.MustLoginToRegistry())
```

Comment:
instead of calling `clientOpts.MustLoginToRegistry()` which would perform the check every time, it could be better to extract it to a boolean before and then reuse that wherever requried

Reply:
I believe Hidde propose this method for better readability. I feel what you propose is like reverting back. I'd like to keep it as is, even though a bit less efficient.